src/juxt/site/alpha/graphql.clj

```diff
@@ -438,6 +440,29 @@
         (log/debugf "defaulting to nil, type-ref is %s" type-ref)
         nil))))
 
+(def validation-template
+  {"minlen" 1
+   "maxlen" 10})
+
+(defn invalid-arguments?
+  [{:keys [argument-values field]}]
+  (let [[k v] (apply vec argument-values)
+        k (keyword k)
+        validation-directive (some-> field
+                                     (get-in [::schema/directives-by-name
+                                              "site"
+                                              ::g/arguments
+                                              "validation"])
+                                     (get k)
+                                     ;; TODO: overwrite it with system-level config
+                                     (selmer/render validation-template)
```

Comment:
I think you should do this in this PR, extract the config used on line 91 into a function that takes the 'opts' map we pass around everywhere and returns the {"type" type-k... map given to selmer/render